crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/mod.rs

```diff
@@ -0,0 +1,168 @@
+use std::marker::PhantomData;
+
+use blockifier::execution::{
+    deprecated_syscalls::DeprecatedSyscallSelector,
+    execution_utils::{stark_felt_from_ptr, ReadOnlySegment},
+    syscalls::{
+        hint_processor::{read_felt_array, SyscallExecutionError, SyscallHintProcessor},
+        SyscallRequest, SyscallResponse, SyscallResponseWrapper, SyscallResult,
+    },
+};
+
+use cairo_felt::Felt252;
+use cairo_vm::{
+    types::relocatable::Relocatable,
+    vm::{errors::hint_errors::HintError, vm_core::VirtualMachine},
+};
+use conversions::{FromConv, IntoConv};
+use num_traits::ToPrimitive;
+use runtime::{ExtendedRuntime, ExtensionLogic, SyscallHandlingResult, SyscallPtrAccess};
+use starknet_api::{core::ContractAddress, hash::StarkFelt};
+
+use crate::state::{BlockifierState, CheatnetState};
+
+use crate::runtime_extensions::call_to_blockifier_runtime_extension::{
+    execution::cheated_syscalls::SingleSegmentResponse,
+    rpc::{call_contract, CallContractFailure, CallContractOutput, CallContractResult},
+};
+
+use super::io_runtime_extension::IORuntime;
+
+pub mod execution;
+pub mod panic_data;
+pub mod rpc;
+
+pub struct CallToBlockifierExtension<'a> {
+    pub lifetime: &'a PhantomData<()>,
+}
+
+pub type CallToBlockifierRuntime<'a> = ExtendedRuntime<CallToBlockifierExtension<'a>>;
+
+impl<'a> ExtensionLogic for CallToBlockifierExtension<'a> {
+    type Runtime = IORuntime<'a>;
+
+    fn override_system_call(
+        &mut self,
+        selector: DeprecatedSyscallSelector,
+        vm: &mut VirtualMachine,
+        extended_runtime: &mut IORuntime<'a>,
+    ) -> Result<SyscallHandlingResult, HintError> {
+        match selector {
+            DeprecatedSyscallSelector::CallContract => {
```

Comment:
We could add a comment that we do it this way to change the execution context (I guess mainly the vm)